client/src/components/AddMates/AddMatesPage.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Button,
+  Col,
+  DatePicker,
+  Divider,
+  Form,
+  Input,
+  Layout,
+  Row,
+  Space,
+  Typography,
+} from "antd";
+import React, { FC, useState } from "react";
+import { Prompt } from "react-router";
+
+import { AddMateFormType, AddMateType } from "../types";
+import AddMateCard from "./AddMateCard";
+
+const { Title } = Typography;
+const { Footer } = Layout;
+
+const AddMatesPage: FC = () => {
+  const [form] = Form.useForm<AddMateFormType>();
+  const [mates, setMates] = useState<AddMateType[]>([]);
+
+  const onAddMate = async ({ name, lastSeen }: AddMateFormType) => {
+    setMates((prevArray) => [
+      ...prevArray,
+      { name, lastSynced: lastSeen.toDate(), ts: new Date().getTime() },
+    ]);
+    form.resetFields();
+  };
+
+  const removeMate = (mate: AddMateType) => {
+    setMates((mateArr) =>
+      mateArr.filter(
+        (m8) =>
+          !(
+            m8.name === mate.name &&
+            m8.lastSynced === mate.lastSynced &&
+            m8.ts === mate.ts
+          )
+      )
+    );
+  };
+
+  const handleAddMates = () => {
```

Comment:
maybe rename to something like submitMates to make it less likely to be confused with onAddMates

Reply:
renamed everything to `NewMate`